data/publiccloud/terraform/openstack.tf

```diff
@@ -0,0 +1,106 @@
+terraform {
+  required_providers {
+    openstack = {
+      version = ">= 0.14.0"
+      source = "terraform-provider-openstack/openstack"
+    }
+    random = {
+      version = "= 3.1.0"
+      source = "hashicorp/random"
+    }
+    external = {
+      version = "= 2.1.0"
+      source = "hashicorp/external"
+    }
+  }
+}
+
+provider "openstack" {
+    cloud = "mycloud"
+}
+
+variable "name" {
+    default = "openqa-vm"
+}
+
+variable "keypair" {
+    description = "Public Key to be injected to the instance"
+    type        = string
+    default     = ""
+}
+
+variable "image_id" {
+    description = "Image ID to boot the instance"
+    type        = string
+    default     = ""
+}
+
+variable "secgroup" {
+    description = "Security group to be used"
+    type        = string
+    default     = "icmp_ssh"
+}
+
+variable "type" {
+    description = "Flavor (cpu, ram) to boot the instance"
+    type        = string
+    default     = "m1.small"
+}
+
+variable "instance_count" {
+    default = "1"
+}
+
+variable "tags" {
+    type = map(string)
+    default = {}
```

Comment:
maybe put same tags as we have for other providers ?

Reply:
I'll try to push it to metadata: https://registry.terraform.io/providers/terraform-provider-openstack/openstack/latest/docs/resources/compute_instance_v2
```
metadata = {
    this = "that"
  }
```

Reply:
Not sure how to push that information... 
```
# timeout 1800 terraform plan -no-color -var 'secgroup=icmp_ssh' -var 'keypair=openqa1644942202_0' -var 'image_id=f1f61591-d2fe-4263-81ef-117fbbe64655' -var 'instance_count=1' -var 'type=m1.small' -var 'region=CustomRegion' -var 'name=jlausuch-vm' -var 'tags={"openqa_ttl":8640300,"openqa_var_JOB_ID":5592,"openqa_var_NAME":"00005592-sle-15-SP3-JeOS-for-OpenStack-Cloud-x86_64-Build4.7.10-jeos-check_boot@64bit-virtio-vga"}' -out myplan; echo JUn31-$?-

Error: Incorrect attribute value type

  on plan.tf line 88, in resource "openstack_compute_instance_v2" "openqa_instance":
  88:     metadata = {
  89:         tags = var.tags
  90:     }

Inappropriate value for attribute "metadata": element "tags": string required.
```

`var.tags` is actually a json, and metadata expects a list of key/value ... so, I'm not sure how to inject that information into the VM without touching the code much... I will skip it for now and investigate further.

Reply:
anyway this is something which is needed by PCW tracking images - and this feature is not done yet so : 
1. we don't know if it actually will be implemented ever :) 
2. if it will we don't know how exactly it will be done ( maybe without tags ) 
so yeah better to remove it for now

Reply:
Yes, I would keep it as a future improvement to be added when we really need it